### PR TITLE
DOC-5369: Cutting the v22.2 docs

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-5369

### CODE REVIEW

File: _config_base.yml
Comment (mikeCRL) [inline _config_base.yml]: What's the impact of including v22.2 generally, and specifically `v22.2-alpha.1` prior to its release?

Also, not sure if related, but I am seeing an empty v22.2 section here
https://deploy-preview-14837--cockroachdb-docs.netlify.app/docs/releases/index.html

Shouldn't we disable that until the alpha release? Should we be moving some steps to the first alpha release notes PR (or a separate PR after this one but before that one)?

**Review by andf-crl — APPROVED** [review]
LGTM after both a diff spot check and a staged HTML spot check. Especially looked for:
- remote_includes
- upgrade guides
- references into and out of out-of-band cockroachcloud dir
- releases stub page and all releases listing (thanks for clarifying that v22.2 is expected to be missing from `releases > CockroachDB` sidebar until first alpha)

Everything looks great -- fantastic work! Let 's get another set of eyes on this though before merge.

Also see DOC-5369 linked tickets for follow-up tasks post-merge (I'll do 'em).

File: v22.2/security-reference/scram-authentication.md
Comment (mikeCRL) [inline v22.2/security-reference/scram-authentication.md]: @MichaelTrestman Do we need a similar but revised note for the v22.2 docs? If a customer were on an upgrade path from v21.2, would't this still be true, but need slight rewording, e.g.

> If you are running a version of CockroachDB prior to v22.1, SCRAM authentication can only be used after an upgrade to v22.1 has been finalized.

Comment (mikeCRL) [reply v22.2/security-reference/scram-authentication.md]: Thanks, Andrew. BTW, just fixed my blockquote - I meant for the last version number there to be "v21.2".

File: _config_base.yml
Comment (nickvigilante) [reply _config_base.yml]: That's a good note. I will update the releases index to suppress any versions that do not contain a release.

If we do not include this in the config, then the main problem we'll face is we won't see v22.2 in the version switcher. Other links may also not resolve correctly and there may be blank bits of text that use `page.version.version` anywhere in the page. I think it's necessary to keep it.

I can instead use `v22.2.0 (unreleased)` or something similar to indicate that it's currently not released. What do you think?

File: _data/versions.csv
Comment (nickvigilante) [reply _data/versions.csv:13]: Adding this helped suppress some errors within `upgrade-cockroach-version.md` and `upgrade-cockroachdb-kubernetes.md `. I was able to correct that temporarily by adding the `{% if rd %}` syntax within those files, but it's not necessary if we just include the row for `v22.2`.

File: v22.2/upgrade-cockroach-version.md
Comment (nickvigilante) [reply v22.2/upgrade-cockroach-version.md:69]: Backward-incompatible changes and deprecated features are not populated until the GA release notes are pushed. The logic here to hide the anchor is necessary until that time to suppress any errors fro htmltest.

File: v22.2/security-reference/scram-authentication.md
Comment (nickvigilante) [reply v22.2/security-reference/scram-authentication.md]: Yep, since upgrading to v22.1 is a required step to upgrade to v22.2 and above, we figured that note was no longer necessary.

File: _data/versions.csv
Comment (nickvigilante) [reply _data/versions.csv:13]: Also, we don't update the release support policy table until we have tangible release dates, which doesn't happen until GA. I wouldn't expect that page to update just yet.

File: v22.2/security-reference/scram-authentication.md
Comment (MichaelTrestman) [reply v22.2/security-reference/scram-authentication.md]: Yup, I concur that the note is no longer needed!

**Comment by nickvigilante** [comment]
@mikeCRL I fixed some issues with the releases index and I undid the change to `upgrade-cockroach-version.md` and `upgrade-cockroachdb-kubernetes.md`. There are 2 outstanding comments left. Hoping to get your final OK on those before I merge.

File: _config_base.yml
Comment (mikeCRL) [reply _config_base.yml]: > I can instead use v22.2.0 (unreleased) or something similar to indicate that it's currently not released. What do you think?

Do you mean that you'd change the value of `name` and `version` to something that's easier for us to understand, but it won't actually be shown on site? If so, that's fine by me. Maybe just `v22.2-alpha.1 (unreleased)` so the next RN PR can just drop 'unreleased'?

Comment (nickvigilante) [reply _config_base.yml]: I don't fully understand why we split name and version and keep them as the same thing, but whatever we set as `version` is the text that is shown directly in the version switcher drop down.

I'll take a separate follow-up action to understand the version plug-in and why it's built the way it's built.

What should we display publicly for end users?

Comment (mikeCRL) [reply _config_base.yml]: To recap, before alpha 1 is released, we need reasonable text for the dropdown re. what version you're seeing. Maybe

`v22.2 (dev / pre-alpha)` ?

Comment (mikeCRL) [reply _config_base.yml]: Edit: After discussion re. limitations, `v22.2 (dev)` is fine. Thanks!

**Review by mikeCRL — APPROVED** [review]
Approving, pending update of version drop-down text to `v22.2 (dev)`

**Pull request merged**
